### ISSUE TITLE
[KafkaStreams Binder] Customizing StreamsBuilder with `KafkaStreamsInfrastructureCustomizer` does not work

### ISSUE THREAD

**Issue by m-kay** [spring-cloud/spring-cloud-stream#2857]
I have an application where I want to register a global state store for every kafka-streams binding before starting the application and therefore I need to configure the `StreamsBuilder`. The `KafkaStreamsInfrastructureCustomizer`s signature looks perfect for that and with the `StreamsBuilderFactoryBeanConfigurer` I can also register it, however the [AbstractKafkaStreamsBinderProcessor](
https://github.com/spring-cloud/spring-cloud-stream-binder-kafka/blob/31ea10683422d38d8185c6b236b8954200562b77/spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/AbstractKafkaStreamsBinderProcessor.java#L343) is indirectly calling  the infrastructure customizer by registing the `StreamsBuilder` in the bean registry before `StreamsBuilderFactoryBeanConfigurer#configure` is called.

This is the way I tried to register my customizer:
```
    @Bean
    fun streamsBuilderFactoryBeanConfigurer(): StreamsBuilderFactoryBeanConfigurer {
        return StreamsBuilderFactoryBeanConfigurer {
            it.setInfrastructureCustomizer(object: KafkaStreamsInfrastructureCustomizer{
                override fun configureBuilder(builder: StreamsBuilder) {
                    print("configuring streams builder")
                }

                override fun configureTopology(topology: Topology) {
                    print("configuring topology")
                }
            })
        }
    }
```

**Comment by sobychacko** [comment]
@m-kay Thanks for reporting this issue. Fixed via the commit above. Which version of Spring Cloud Stream are you using? We can back-port the changes to the `4.0.x` branch.

**Comment by m-kay** [comment]
I'm using the spring-cloud version 2022.0.4 which uses spring-cloud-stream 4.0.4. So yes it would be nice to have that backported and available in the next spring-cloud release.
